crates/turborepo-lib/src/cli/mod.rs

```diff
@@ -612,7 +612,7 @@ fn validate_graph_extension(s: &str) -> Result<String, String> {
 ])]
 pub struct RunArgs {
     /// Override the filesystem cache directory.
-    #[clap(long)]
+    #[clap(long, env = "TURBO_CACHE_DIR")]
```

Comment:
This currently accepts `TURBO_CACHE_DIR= turbo build` which leads to some unwanted behavior. We especially need to cover this case as this is what Docker sets `ARG`s to if they aren't provided.

We need to use an alternative value parser than the parser for `Utf8PathBuf` as `""` is an valid path, but will probably result in unwanted behavior.

Some doc pointers:
- [Arg::env docs](https://docs.rs/clap/latest/clap/struct.Arg.html#method.env)
- [value_parser docs](https://docs.rs/clap/latest/clap/struct.Arg.html#method.value_parser)

I can help more if needed.

Reply:
Ah I just checked again and the `--cache-dir` issue itself has the same issue (It does throw on `turbo build --cache-dir`, but allows `turbo build --cache-dir=` or `turbo build --cache-dir ''). 

Let me add that validation, get that merged, and then we can merge this.

Reply:
Okay. Once https://github.com/vercel/turbo/pull/7630 gets merged, this will be fine to merge. Thanks for the PR!

Reply:
Awesome, thank you very much for your help @chris-olszewski :)